aiida/engine/processes/calcjobs/calcjob.py

```diff
@@ -148,12 +148,12 @@ def define(cls, spec):
             help='Set the quality of service to use in for the queue on the remote computer')
         spec.input('metadata.options.withmpi', valid_type=bool, default=False,
             help='Set the calculation to use mpi',)
-        spec.input('metadata.options.mpirun_extra_params', valid_type=(list, tuple), default=[],
+        spec.input('metadata.options.mpirun_extra_params', valid_type=(list, tuple), default=list,
```

Comment:
What is there to understand though? They won't look in the implementation I imagine and with the `ProcessBuilder` auto help it will show `default=list`, which seems fine no? Anyway, I wouldn't know how to solve this otherwise, because putting `None` will break backwards-compatibility.

Reply:
This might sound silly, but maybe we could use `list()` and `dict()`? At least for me that makes it more clear that you are creating an empty list/dict and not passing the actual class.

Reply:
the problem is though, that as soon as you add the parentheses you are _calling_ the callable and so you are instantiating it. That is exactly what we don't want to do. The builint `list` _is_ a callable and calling it like `list()` creates an instance of that type. This is just the way python works.

Reply:
I think part of why this might be hard to catch for new users is naming: The ``default`` argument essentially combines the concepts of a default and a ``default_factory`` (in the sense of ``defaultdict`` or ``dataclasses.field``).

In my opinion the interface would be cleaner if the two concepts were separated, but for backwards compatibility I would only change this _very_ slowly, if at all.